File: pat/1005.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;
string map[10] = {"zero", "one", "two", "three", "four", "five",
    "six", "seven", "eight", "nine"};

int main ()
{
    string s;
    int sum = 0;
    vector<string> res;
    cin >> s;
    if (s == "0") {
        cout << "zero" << endl;
        return 0;
    }
    for (int i = 0; i < s.size(); i ++ ) {
        sum += s[i] - '0';
    }
    while (sum > 0) {
        res.push_back(map[sum % 10]);
        sum /= 10;
    }
    for (int i = res.size() - 1; i >= 0; i --) {
        cout << res[i];
        if (i != 0) {
            cout << " ";
        }
    }
    cout << endl;
    return 0;
}

File: pat-go/1005.go
package main

import (
	"fmt"
)

func main() {
	var input string
	fmt.Scanln(&input)
	var sum int = 0
	for i := 0; i < len(input); i++ {
		sum += int(input[i] - '0')
	}
	var words = [10]string{"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"}
	arr := make([]int, 0)
	for true {
		arr = append(arr, sum%10)
		sum /= 10
		if sum == 0 {
			break
		}
	}
	for i := len(arr) - 1; i >= 0; i-- {
		fmt.Printf(words[arr[i]])
		if i != 0 {
			fmt.Printf(" ")
		}
	}
	fmt.Printf("\n")
}
